### PR TITLE
Chore/maintenance

### PR DESCRIPTION
**Pull request by FabianMeul**
## What

- Monorepo refactor

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,43 +42,43 @@
     "@semantic-release/git": "^10.0.1",
     "@semantic-release/github": "^8.0.4",
     "@semantic-release/npm": "^9.0.0",
-    "@types/jest": "^27.5.1",
-    "@types/react": "^18.0.14",
```
Comment (JaccoGoris) [inline package.json:46]: For the record: not a fan of locking types

File: package.json [hunk 2]
```diff
@@ -42,43 +42,43 @@
...
     "eslint-config-prettier": "^8.5.0",
-    "eslint-import-resolver-typescript": "^2.7.1",
+    "eslint-import-resolver-typescript": "3.2.4",
     "eslint-plugin-eslint-comments": "^3.2.0",
     "eslint-plugin-import": "^2.26.0",
     "eslint-plugin-jest": "^26.5.3",
     "eslint-plugin-jest-formatting": "^3.1.0",
     "eslint-plugin-jsx-a11y": "^6.5.1",
     "eslint-plugin-no-secrets": "^0.8.9",
-    "eslint-plugin-prettier": "^4.1.0",
```
Comment (JaccoGoris) [inline package.json:70]: locked version?

File: package.json [hunk 3]
```diff
@@ -42,43 +42,43 @@
...
     "typescript": "^4.7.4"
   },
   "dependencies": {
     "@react-native-community/eslint-config": "^3.0.3",
-    "@typescript-eslint/eslint-plugin": "^5.29.0",
-    "@typescript-eslint/parser": "^5.29.0",
+    "@typescript-eslint/eslint-plugin": "^5.30.5",
+    "@typescript-eslint/parser": "^5.30.5",
     "eslint-config-prettier": "^8.5.0",
-    "eslint-import-resolver-typescript": "^2.7.1",
```
Comment (JaccoGoris) [inline package.json:63]: locked version?

File: package.json [hunk 4]
```diff
@@ -34,53 +23,45 @@
...
-    "eslint-plugin-sonarjs": "^0.13.0",
-    "eslint-plugin-unicorn": "^42.0.0",
-    "eslint-plugin-woke": "^1.0.1"
-  },
-  "peerDependencies": {
-    "eslint": "8.x",
-    "jest": "27",
-    "prettier": "2.x",
-    "react": "17 - 18",
-    "typescript": "4.x"
```
Comment (JaccoGoris) [inline package.json:84]: why are the peer deps gone?

File: packages/@bothrs-eslint-config-jest/src/config.spec.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    expect(async () => {
+      const files = path.resolve('./specs/samples/vanilla.js')
+
+      await eslint.lintFiles([
+        files
+      ])
+
+      // const results = ESLint.getErrorResults(result)
+
+      // console.log(result[0].messages[0], results)
```
Comment (JaccoGoris) [inline packages/@bothrs-eslint-config-jest/src/config.spec.ts:24]: comments

File: packages/@bothrs-eslint-config-jest/src/config.spec.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  it('Handles an error from each of the configured plugins', async () => {
+
+    const files = path.resolve('./specs/samples/vanilla.js')
+
+    const result = await eslint.lintFiles([
+      files
+    ])
+
+    const eslintResult = result[0]
+    // const prettierError = eslintResult.messages.find(( value ) => value.ruleId === 'prettier/prettier')
```
Comment (JaccoGoris) [inline packages/@bothrs-eslint-config-jest/src/config.spec.ts:37]: still needed?

File: packages/@bothrs-eslint-config-next/package.json [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    "build": "esbuild src/config.ts --bundle --outfile=index.js --format=cjs",
+    "lint": "eslint specs/samples/vanilla.js",
+    "test": "jest",
+    "test:coverage": "jest --coverage",
+    "test:spec": "eslint --format json",
+    "test:verbose": "jest --verbose"
+  },
+  "devDependencies": {
+    "@next/eslint-plugin-next": "^12.3.1"
+  },
```
Comment (JaccoGoris) [inline packages/@bothrs-eslint-config-next/package.json:43]: would make this a peerdep too

File: packages/@bothrs-eslint-config-next/src/config.spec.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    expect(async () => {
+      const files = path.resolve('./specs/samples/vanilla.js')
+
+      await eslint.lintFiles([
+        files
+      ])
+
+      // const results = ESLint.getErrorResults(result)
+
+      // console.log(result[0].messages[0], results)
```
Comment (JaccoGoris) [inline packages/@bothrs-eslint-config-next/src/config.spec.ts:25]: comments

File: packages/@bothrs-eslint-config-node/src/config.spec.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    expect(async () => {
+      const files = path.resolve('./specs/samples/vanilla.js')
+
+      await eslint.lintFiles([
+        files
+      ])
+
+      // const results = ESLint.getErrorResults(result)
+
+      // console.log(result[0].messages[0], results)
```
Comment (JaccoGoris) [inline packages/@bothrs-eslint-config-node/src/config.spec.ts:25]: comments

File: packages/@bothrs-eslint-config-react-native/package.json [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+  },
+  "devDependencies": {
+    "@types/react": "^18.0.21"
+  },
+  "dependencies": {
+    "@bothrs/eslint-config": "^2.0.0",
+    "@bothrs/eslint-config-react": "^2.0.0",
+    "@react-native-community/eslint-config": "^3.1.0",
+    "eslint-plugin-react-native": "^4.0.0",
+    "eslint-plugin-react-native-a11y": "^3.2.1"
```
Comment (JaccoGoris) [inline packages/@bothrs-eslint-config-react-native/package.json:49]: probably also want these as peer deps so are not in the project twice

File: packages/@bothrs-eslint-config-react/package.json [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    "test:verbose": "jest --verbose"
+  },
+  "devDependencies": {
+    "@types/react": "^18.0.21"
+  },
+  "dependencies": {
+    "@bothrs/eslint-config": "^2.0.0",
+    "eslint-plugin-jsx-a11y": "^6.6.1",
+    "eslint-plugin-react": "^7.31.9",
+    "eslint-plugin-react-hooks": "^4.5.0"
```
Comment (JaccoGoris) [inline packages/@bothrs-eslint-config-react/package.json:48]: peerdeps?

File: packages/@bothrs-eslint-config-react/src/config.spec.ts [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+    expect(async () => {
+      const files = path.resolve('./specs/samples/Component.jsx')
+
+      await eslint.lintFiles([
+        files
+      ])
+
+      // const results = ESLint.getErrorResults(result)
+
+      // console.log(result[0].messages[0], results)
```
Comment (JaccoGoris) [inline packages/@bothrs-eslint-config-react/src/config.spec.ts:25]: logs

File: packages/@bothrs-eslint-config-typescript/package.json [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+    "test": "jest",
+    "test:coverage": "jest --coverage",
+    "test:spec": "eslint --format json",
+    "test:verbose": "jest --verbose"
+  },
+  "dependencies": {
+    "@bothrs/eslint-config": "^2.0.0",
+    "@typescript-eslint/eslint-plugin": "^5.39.0",
+    "@typescript-eslint/parser": "^5.39.0",
+    "eslint-import-resolver-typescript": "^3.5.1"
```
Comment (JaccoGoris) [inline packages/@bothrs-eslint-config-typescript/package.json:45]: peerdeps?

File: packages/@bothrs-eslint-config-typescript/src/config.spec.ts [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+  it('Initializes without errors', () => {
+    expect(async () => {
+      const files = path.resolve('./specs/samples/vanilla.ts')
+
+      await eslint.lintFiles([files])
+
+      // const results = ESLint.getErrorResults(result)
+
+      // console.log(result[0].messages[0], results)
```
Comment (JaccoGoris) [inline packages/@bothrs-eslint-config-typescript/src/config.spec.ts:21]: logs

File: package.json (hunk 4 above)
Comment (FabianMeul) [reply package.json:84]: The packages are in the `packages/*`-folder.
The monorepo itself has no use for peerDependencies.

Comment (JaccoGoris) [reply package.json:84]: this makes a lot of sense

**Review by JaccoGoris — APPROVED** [review]

**Pull request merged**